knowledge-portal/source/source-pipeline-checks.py

```diff
@@ -0,0 +1,119 @@
+import os
+import sys
+from glob import glob
+
+# Search for manifest file
+manifest_file = glob("docs-manifest.txt")[0]
+
+# Check if folder copy scenario; if yes, get all files in the folder recursively
+def check_for_folders(source_paths):
+    folder_mappings = {}
+    # Folder mappings expanded to their corresponding recursive file mappings
+    for source_path in source_paths:
+        # Check if source_path is a folder
+        if os.path.exists(source_path):
+            if os.path.isdir(source_path):
+                if source_path.endswith("/"):
+                    mycwd = os.getcwd()
+                    os.chdir(source_path)
+                    files = glob('**/*.*', recursive=True)
+                    #files = [f"{source_path}{file}" for file in files]
+                    os.chdir(mycwd)
+                    folder_mappings[source_path] = files
+                else:
+                    sys.exit(f"Check failed! This folder: {source_path} path doesn't end with (/)")
+        else:
+            sys.exit(f"Check failed! This file/folder: {source_path} mentioned in manifest file doesn't exist or you have not ended your folder copy with (/)")
+    return folder_mappings
```

Comment:
Quick refactoring for easier readability:

```python
```

Reply:
The `glob` function being used is a focused implementation within the `pathlib` library, that's it :)

Reply:
Also, another note: You could experiment replacing the `if xyz: self.exit()` checks internally with an `assert()`. A point to note though is that `assert`s usually have a different program-exit-code than `sys.exit()` (0 by default?), and so you'll have to see its impact on the gitlab runner

Reply:
I have used my function itself for now.